modules/spring/src/test/java/org/apache/ignite/cache/store/jdbc/CacheJdbcPojoWriteBehindHangStoreWithCoalescingTest.java

```diff
@@ -0,0 +1,148 @@
+        cleanPersistenceDir();
+
+        ignite = Ignition.start("modules/spring/src/test/config/jdbc-write-behind-coalescing.xml");
+
+        ignite.cluster().active(true);
+
+        super.beforeTest();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        Ignition.stop(ignite.configuration().getIgniteInstanceName(), true);
+
+        cleanPersistenceDir();
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testNonCoalescingIncrementing() throws Exception {
+        try {
+            Connection conn = DriverManager.getConnection(DFLT_CONN_URL, "sa", "");
+
+            Statement stmt = conn.createStatement();
+
+            stmt.executeUpdate("DROP TABLE IF EXISTS TEST_CACHE");
+
+            stmt.executeUpdate("CREATE TABLE TEST_CACHE (" +
+                " VALUE2 INTEGER PRIMARY KEY," +
+                " VALUE1 VARCHAR(50)," +
+                " VALUE3 DATE" +
+                ")"
+            );
+
+            conn.commit();
+
+            U.closeQuiet(stmt);
+
+            U.closeQuiet(conn);
+        } catch (SQLException ex) {
+
+        }
+
+        IgniteCache<Integer, TestPogo> cache = ignite.cache("TEST_CACHE");
+
+        AtomicInteger t1Count = new AtomicInteger(10);
+        AtomicInteger t2Count = new AtomicInteger(10);
+
+        Thread t1 = new Thread(new Runnable() {
+            @Override public void run() {
+                try {
+                    while (t1Count.get() > 0) {
+                        for (int i = 0; i < 5000; i++)
+                            cache.put(i, new TestPogo("ORIGIN" + i, i, new java.sql.Date(new java.util.Date().getTime())));
+
+                        t1Count.decrementAndGet();
+                    }
+                } catch (CacheException e) {
+                    //ignore
+                }
+            }
+        });
+
+        Thread t2 = new Thread(new Runnable() {
+            @Override public void run() {
+                try {
+                    while (t2Count.get() > 0) {
+                        for (int i = 0; i < 5000; i++)
+                            cache.put(i, new TestPogo("UPDATE" + i, i, new java.sql.Date(new java.util.Date().getTime())));
+
+                        try {
+                            U.sleep(500);
+                        }
+                        catch (IgniteInterruptedCheckedException e) {
+                            e.printStackTrace();
+                        }
+
+                        t2Count.decrementAndGet();
+                    }
+                } catch (CacheException e) {
+                    //ignore
+                }
+            }
+        });
+
+        t1.start();
+
+        t2.start();
+
+        //every write to cache store will take more then 10 seconds. t1 should be completed before 10 seconds.
+        U.sleep(10000);
+
+        assertEquals(0, t1Count.get());
```

Comment:
Please add a consistency check that the cache contains the same values as the underlying store.

Reply:
done